scripts/west_commands/fetchers/http.py

```diff
@@ -16,5 +18,14 @@ def schemes(cls):
 
     def fetch(self, url, path):
         log.dbg(f'HTTPFetcher fetching {url} to {path}')
-        resp = requests.get(url)
-        open(path, "wb").write(resp.content)
+        try:
+            resp = requests.get(url)
+            resp.raise_for_status()  # Raises an HTTPError for bad status codes (4xx or 5xx)
+        except requests.exceptions.HTTPError as e:
+            log.err(f'HTTP error occurred: {e}')
+            sys.exit(os.EX_NOHOST)
+        except requests.exceptions.RequestException as e:
+            log.err(f'An error occurred: {e}')
+            sys.exit(os.EX_DATAERR)
+        else:
+            open(path, "wb").write(resp.content)
```

Comment:
No need to put it in an `else`, also prefer using a context

```suggestion
        with open(path, "wb") as f:
            f.write(resp.content)
```